src/Nethermind/Nethermind.Network/P2P/RateLimitedPacketAcceptanceStrategy.cs

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Network.Rlpx;
+
+namespace Nethermind.Network.P2P;
+
+public class RateLimitedPacketAcceptanceStrategy : IPacketAcceptanceStrategy
+{
+    private readonly int _byteLimit;
+    private readonly TimeSpan _throttle;
+
+    private int _bytesAccepted;
+    private DateTime _timestamp;
```

Comment:
Instead of `DateTime`, it's possible to use `TimeSpan` with a `Stopwatch`. I'm not sure which is best though.

Reply:
Bonus point if you can encapsulate the ratelimit logic to its own class. For example, there is a `RateLimiter` class but that is for observation per second, not for bytes.